legal-api/src/legal_api/resources/v2/business/business_documents.py

```diff
@@ -48,44 +48,44 @@ def get_business_documents(identifier: str, document_name: str = None):
             HTTPStatus.UNAUTHORIZED,
         )
 
-    legal_entity = LegalEntity.find_by_identifier(identifier)
+    business = business_service.fetch_business(identifier)
 
-    if not legal_entity:
+    if not business:
         return (
             jsonify(message=get_error_message(ErrorCode.MISSING_BUSINESS, **{"identifier": identifier})),
             HTTPStatus.NOT_FOUND,
         )
 
     if not document_name:
-        return _get_document_list(legal_entity)
+        return _get_document_list(business)
 
-    err = validate_document_request(document_name, legal_entity)
+    err = validate_document_request(document_name, business)
     if err:
         response_message = {"errors": err.msg}
         return jsonify(response_message), err.code
 
     if document_name:
         if "application/pdf" in request.accept_mimetypes:
-            return BusinessDocument(legal_entity, document_name).get_pdf()
+            return BusinessDocument(business, document_name).get_pdf()
         elif "application/json" in request.accept_mimetypes:
-            return BusinessDocument(legal_entity, document_name).get_json()
+            return BusinessDocument(business, document_name).get_json()
     return {}, HTTPStatus.NOT_FOUND
 
 
-def _get_document_list(legal_entity):
+def _get_document_list(business):
     """Get list of business documents."""
     base_url = current_app.config.get("LEGAL_API_BASE_URL")
     base_url = base_url[: base_url.find("/api")]
-    doc_url = url_for("API2.get_business_documents", **{"identifier": legal_entity.identifier, "document_name": None})
+    doc_url = url_for("API2.get_business_documents", **{"identifier": business.identifier, "document_name": None})
     business_documents = ["summary"]
     documents = {"documents": {}}
 
     for doc in business_documents:
         documents["documents"][doc] = f"{base_url}{doc_url}/{doc}"
 
-    if legal_entity.entity_type == LegalEntity.EntityTypes.COOP.value:
+    if business.is_legal_entity == LegalEntity.EntityTypes.COOP.value:
```

Comment:
L86 should be `if business.entity_type == LegalEntity.EntityTypes.COOP.value:`

Reply:
fixed